doc/sphinx_source/tutorials/module.rst

```diff
@@ -0,0 +1,234 @@
+Writing an Eggdrop Module
+=========================
+
+An Eggdrop module is a piece of C code that can be loaded (or unloaded) onto the core Eggdrop code. A module differs from a Tcl script in that modules must be compiled and then loaded, whereas scripts can be edited and loaded directly. Importantly, a module can be written to create new Eggdrop-specific Tcl commands and binds that a user can then use in a Tcl script. For example, the server module loaded by Eggdrop is what creates the "jump" Tcl command, which causes tells the Eggdrop to jump to the next server in its server list.
+
+There are a few required functions a module must perform in order to properly work with Eggdrop
+
+Module Header
+-------------
+
+A module should include license information. This tells other open source users how they are allowed to use the code. Eggdrop uses GPL 2.0 licensing, and our license information looks like this::
+
+  /*
+  * This program is free software; you can redistribute it and/or
+  * modify it under the terms of the GNU General Public License
+  * as published by the Free Software Foundation; either version 2
+  * of the License, or (at your option) any later version.
+  *
+  * This program is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program; if not, write to the Free Software
+  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+  */
+
+Required Code
+-------------
+
+For this section, you don't necessarily need to understand what it is doing, but this code is required for a module to function.
+
+You'll next want to name your module::
+
+  #define MODULE_NAME "woobie"
+
+Declare your own function tables (you don't need to understand this part; you just need to copy/paste it)::
+
+  #undef global
+  static Function *global = NULL, *server_funcs = NULL;
+  EXPORT_SCOPE char *woobie_start();
+
+Next are two memory-reporting functions used by the core Eggdrop .status and .module commands::
+
```

Comment:
Placeholder, in reality it should return the number of bytes that are dynamically allocated and not free'd